challenges-14-plus.yml

```diff
@@ -0,0 +1,472 @@
+    `app_pizzas` that we want to prefetch on every page so that such page is
+    always ready to be loaded instantly if the user wants to go there.
+    
+    Assuming we have a `{% block metas %}{% endblock %}` that we can override. What
+    is the native way in which we can do this?
+  explanation: |
+    `<link rel="prefetch">` is a useful tool that you can leverage with Turbo Drive
+    when you need to make certain pages readily available to the user.
+    
+    In our example, we infer that we are adding this prefetch link on every page. Make
+    sure you have this in mind, because if you add a prefetch link in a *specific* page,
+    this won't stay in the `<head>` element for other pages!
+  features:
+    multiple_choice:
+      a: |
+        ```twig
+        <link rel="prefetch" src="{{ path('app_pizzas') }}">
+        ```
+      b: |
+        ```twig
+        <link rel="prefetch" href="{{ path('app_pizzas') }}">
+        ```
+      c: |
+        ```twig
+        <a rel="prefetch" href="{{ path('app_pizzas') }}"></a>
+        ```
+      d: |
+        ```twig
+        <meta rel="prefetch" href="{{ path('app_pizzas') }}">
+        ```
+    correct: b
+
+lazy_frame:
+  question: |
+    What is a Turbo Frame and what does it do?
+  explanation: |
+    The `<turbo-frame>` HTML element is a special element that allows us
+    to make areas of our web pages dynamic.
+    
+    You can use Turbo Frames if you want navigation to happen inside an
+    area of the page, instead of having the whole page refreshed.
+    
+    Another use is when you want a part of your page to load lazily, once
+    the general structure has loaded.
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          A Turbo Frame is an `<iframe>` element in a page that has extra
+          functionality, like loading pages via ajax calls.
+        b: |
+          Tturbo Frames is a system that allows you to create complex animations
+          in our web pages.
+        c: |
+          A Turbo Frame is an area of our page in which we can load content and
+          update it via Ajax calls that return HTML. It behaves similarly to the
+          old `<iframe>` system.
+        d: |
+          A Turbo Frame is an area of our page that has the ability to stay loaded
+          while other parts of our page change. It is mainly used as navigation bars.
+      correct: c
+
+frames_find_frames:
+  question: |
+    My website has a terribly complex sidebar! Our project manager asked us to extract
+    it to a separate route so that we could load it lazily using Turbo Frames. This
+    work is done! Our route has the name `_app_nav`, and the rendered frame has an id of
+    `app_nav_frame`.
+    
+    How could we load this in our main template?
+  explanation: |
+    It's very important that when a frame is loaded by Turbo, its HTML contains at least
+    one `<turbo-frame>` element with an `id` that we can identify!
```

Comment:
The `src=""` attribute tells Turbo to make an Ajax call to this URL to fetch the frame.
Once it does, it reads the `id` attribute of the original `<turbo-frame>` (`id="app_nav_frame"`)
and finds a `turbo-frame` with that *same* id inside the Ajax response's HTML.